catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipCompression.java

```diff
@@ -130,10 +132,19 @@ private InputStream createZip(
           zipOutputStream.putNextEntry(entry);
           zipOutputStream.write(binaryContent.getByteArray());
           zipOutputStream.closeEntry();
+          totalWritten++;
         } else {
+          notAddedIds.add(metacard.getId());
           LOGGER.debug("Metacard with id [{}] was not added to zip file", metacard.getId());
         }
       }
+      zipOutputStream.finish();
```

Comment:
❓ Interesting were we never finishing the ZIP or was that happening on a different code path not visible on this PR?

Reply:
Don't think we were ever finishing the zip, abel said this addition alone fixed most of the zip export issues